src/harness/reference_models/examples/benchmark/time_dpa.py

```diff
@@ -0,0 +1,257 @@
+"""Benchmark the DPA model timing.
+
+This is a simple simulation of the DPA for purpose of estimating time to run.
+
+Usage:
+  python time_dpa.py
+
+Notes:
+  - multiprocessing facility not tested on Windows (use 1 process if issues)
+  - if warning reported on cached tiles swapping, increase the cache size.
+"""
+# TODO(sbdt): add reporting for memory usage
+from collections import namedtuple
+import multiprocessing
+import time
+
+import cartopy.crs as ccrs
+import matplotlib.pyplot as plt
+import numpy as np
+
+from reference_models.move_list_calc import move_list
+from reference_models.geo import zones
+from reference_models.geo import drive
+from reference_models.geo import utils
+from reference_models.examples import entities
+
+#------------------------------
+# Define simulation parameters
+# - DPA to simulate
+dpa_name = 'east_dpa_7'
+
+# - Number of sites to distribute within a given range of the DPA
+num_sites = 500
+max_dist_cat_a = 160
+max_dist_cat_b = 320
+
+# - Number of protection points in DPA per category.
+# We divide the DPA in 4 buckets: front and back contour, front and back zone
+# The front and back are separeted by the us border with some buffering.
+# Note:  Keep a ratio as expected in final to get accurate timing estimate
+npts_front_dpa_contour = 250    # The mainland facing contour
+npts_back_dpa_contour = 100     # The back contour
+npts_within_dpa_front = 100     # The front zone
+npts_within_dpa_back = 50       # The back zone
+front_usborder_buffer_km = 40   # Front contour defined by the extension of the
+                                # front border.
+
+# - Do we restrict the sites to be within the census-defined urban areas
+do_inside_urban_area = True
+
+# - Ratio of cat B and different catA
+ratio_cat_b = 0.2
+ratio_cat_a_indoor = 0.4
+
+# - Simulation setup
+#   + Number of parallel processes -
+#     do not exceed your workstation parallel capabilities (number of core -2)
+num_processes = 8
```

Comment:
We can use `multiprocessing.cpu_count()` to set the default value here and subtract it by 2, as suggested in the comments?

Reply:
This is a "benchmarking code", so the idea is to run it in some environment with
precise parameters. And extrapolate from that.
What i can do, is that value -1 does what you suggest. But instead of 50% of the resources, I would say 75% ?